include/tvm/relay/pass.h

```diff
@@ -156,7 +175,74 @@ size_t StructuralHash(const Type& type);
  */
 size_t StructuralHash(const Expr& expr);
 
+/*! \brief The hash struct for expressions. */
+struct ExprHash {
+  size_t operator()(const Expr& a) const { return StructuralHash(a); }
+};
+
+/*! \brief The equal comparator for expressions. */
+struct ExprEqual {
+  bool operator()(const Expr& a, const Expr& b) const {
+    return AlphaEqual(a, b);
+  }
+};
+
+/*! \brief A lowered Relay operation.
+ *
+ * A lowered operation is a pair containing the "primitive" function used
+ * to produce the lowered function as well as the lowered function itself.
```

Comment:
This seems should belong to the internal header for now, as codegen is not strictly a pass

Reply:
This pass is exposed to Python for the runtime compiler, all the other exposed operations are in `pass.h`.

Reply:
I get that, on the other hand, I feel we should distinguish from pass(IR->IR) to build(IR->Module), and that might merit a second module, putting it in the internal is just a safe choice for now